Payload/Generator.php

```diff
@@ -2,46 +2,24 @@
 
 namespace Rollbar\Symfony\RollbarBundle\Payload;
 
+use JetBrains\PhpStorm\ArrayShape;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpKernel\Kernel;
 
-/**
- * Class Generator
- *
- * @package Rollbar\Symfony\RollbarBundle\Payload
- */
 class Generator
 {
-    /**
-     * @var ContainerInterface
-     */
-    protected $container;
-
-    /**
-     * @var Kernel
-     */
-    protected $kernel;
+    protected ?Kernel $kernel;
 
-    /**
-     * Generator constructor.
-     *
-     * @param ContainerInterface $container
-     */
-    public function __construct(ContainerInterface $container)
+    public function __construct(protected ContainerInterface $container)
     {
-        $this->container = $container;
-        $this->kernel    = $container->get('kernel');
+        $this->kernel = $container->get('kernel');
     }
 
     /**
      * Get payload a log record.
-     *
-     * @param \Exception $exception
-     *
-     * @return array
      */
-    public function getExceptionPayload($exception): array
+    public function getExceptionPayload(\Exception $exception): array
```

Comment:
I agree that `Throwable` would be a better type for this than `Exception`. Both the `TraceChain` and `TraceItem` classes specify the `Throwable` interface as the types they use for the `__invoke()` methods which are the only uses of `$exception` in the `Generator::getExceptionPayload()` method. This is a great observation! 

I think while updating this I will rename `$exception` to `$throwable` in `Generator::getExceptionPayload()` so that it matches the naming in both `TraceChain` and `TraceItem`.

Reply:
This has been updated.